### DISCUSSION TITLE
请问openwrt-passwall2和openwrt-passwall有什么区别，后期维护计划是什么

### DISCUSSION THREAD

**Discussion by HenTele** [Openwrt-Passwall/openwrt-passwall2#555] [General]
如题，近期发现passwall有一波commit，但是还没有合并到passwall2，想咨询一下开发者大大，后期是会逐步过渡到passwall2然后关闭passwall吗？还是两边都会维护？上上个月才迁移到passwall2，但是最近的passwall2的bug较多，眼看着passwall又在更新了，不知道是不是要再迁移回去？

**Comment by chshouyu** [comment]
作为一个使用者来说，感觉 passwall2 更纯粹更完美，更像是一个 Xray/Sing-box 的 UI
相比于 passwall，passwall2 抛弃了直连、屏蔽等列表，所有的规则只使用 keyword、domain、geosite、geoip 去匹配，非常容易理解和维护
通过自定义分流列表，自定义列表顺序，可以非常轻松的决定哪个域名走直连还是代理，哪个域名走哪个代理节点，有一种完全掌控的感觉

**Reply by Yumega** [reply]
> 作为一个使用者来说，感觉 passwall2 更纯粹更完美

相同的配置passwall 打开国外网页明显比pw2快的多的多

**Reply by Travel2Here** [reply]
> 的多的多

朋友，这个就……哈哈哈了，底层都一样咋可能呢？（xray/singbox的程序+配置文件，两者皆相同）

我个人也是与楼上一样的看法，逻辑相对清晰简单，有完全掌控的可能性。

**Reply by SenkjM** [reply]
> > 的多的多
> 
> 朋友，这个就……哈哈哈了，底层都一样咋可能呢？（xray/singbox的程序+配置文件，两者皆相同）
> 
> 我个人也是与楼上一样的看法，逻辑相对清晰简单，有完全掌控的可能性。

passwall可以利用防火墙提前转发掉一部分流量，理论上的性能表现就是会更好

**Reply by hxlls** [reply]
> > > 的多的多
> > 
> > 
> > 朋友，这个就……哈哈哈了，底层都一样咋可能呢？（xray/singbox的程序+配置文件，两者皆相同）
> > 我个人也是与楼上一样的看法，逻辑相对清晰简单，有完全掌控的可能性。
> 
> passwall可以利用防火墙提前转发掉一部分流量，理论上的性能表现就是会更好

![image](https://github.com/user-attachments/assets/30b69fbd-1235-437e-bbbf-c03d67f1ad18)
pw2也有的
